File: course_android/LaunchMode_FLAG_NEW_TASK/app/src/main/java/com/better/learn/launchmode_flag_new_task/OtherActivity.java
package com.better.learn.launchmode_flag_new_task;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

public class OtherActivity extends BaseAct {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_other);
        findViewById(R.id.main_click).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(OtherActivity.this, MainActivity.class)
                        .setFlags(Intent.FLAG_ACTIVITY_NEW_TASK));
            }
        });
        findViewById(R.id.main_2).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //这种方式 会清除之前的实例到栈顶的Activity，在创建一个新的放入到栈中 并不会调用`onNewIntent()`
                startActivity(new Intent(OtherActivity.this, MainActivity.class)
                        .setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP));
            }
        });
        findViewById(R.id.main_3).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //这种方式 像singleTask，启动主页后，是将之前的主页展现，并移除Activity
                startActivity(new Intent(OtherActivity.this, MainActivity.class)
                        .setFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP | Intent.FLAG_ACTIVITY_CLEAR_TOP));
            }
        });
    }
}

File: course_android/Animation50/app/src/main/java/better/learn/animation50/OtherActivity.kt
package better.learn.animation50

import android.annotation.TargetApi
import android.os.Build
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.transition.Fade
import android.transition.Slide
import android.view.Window

class OtherActivity : AppCompatActivity() {

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        window.requestFeature(Window.FEATURE_CONTENT_TRANSITIONS)
        setContentView(R.layout.activity_other)

        window.enterTransition=Slide()
        window.exitTransition =Fade()
    }
}
